File: 724-find-pivot-index/solution.py
class Solution:
    def pivotIndex(self, nums: list[int]) -> int:
        ridx = len(nums) - 1
        sum_l = []
        sum_r = []
        for idx, i in enumerate(nums):
            if idx == 0:
                sum_l.append(i)
                sum_r.append(nums[ridx])
            else:
                sum_l.append(sum_l[idx-1] + i)
                sum_r.insert(0, sum_r[0] + nums[ridx])
            ridx -= 1

        for i in range(len(nums)):
            if sum_l[i] == sum_r[i]:
                return i
        return -1

File: 724-find-pivot-index/solution.go
package main

func pivotIndex(nums []int) int {
    n := len(nums)
    ridx := n - 1
    l_sum := make([]int, n)
    r_sum := make([]int, n)

    for idx, i := range nums {
        if idx == 0 {
            l_sum[idx] = i
            r_sum[ridx] = nums[ridx]
        } else {
            l_sum[idx] = i + l_sum[idx-1]
            r_sum[ridx] = r_sum[ridx+1] + nums[ridx]
        }
        ridx--
    }

    for i := 0; i < n; i++ {
        if l_sum[i] == r_sum[i] {
            return i
        }
    }
    return -1
}